### PR TITLE
refactor: update application styles

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

[Doorway alignment issue
](https://github.com/metrotranscom/doorway/issues/364)
## Description

A handful of changes to support the updates to the common app in core. I left comments on most of the files to clarify.

## How Can This Be Tested/Reviewed?

Check out the ProgressNav component with the recent updates in [Storybook](https://deploy-preview-132--storybook-bloom-dev.netlify.app/?path=/story/navigation-progress-nav-%F0%9F%9A%A9--default).

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have made any corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added or updated stories if new changes are not captured in Storybook
- [x] New and existing unit tests pass locally with my changes
- [ ] I have exported any new pieces added to ui-components
- [ ] I have documented this change in the changelog, and any breaking changes are well described

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Either review the Storybook preview or pull the changes down locally and test that the acceptance criteria is met
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

On merge, squash commits.

### CODE REVIEW

File: src/forms/DOBField.tsx
Comment (emilyjablonski) [inline src/forms/DOBField.tsx:1]: Removed placeholder text and resolved a linting error

File: src/forms/PhoneField.tsx
Comment (emilyjablonski) [inline src/forms/PhoneField.tsx:1]: This was causing style discrepancies with just this field

File: src/global/forms.scss
Comment (emilyjablonski) [inline src/global/forms.scss:1]: This was causing style discrepancies

File: src/helpers/MultiLineAddress.tsx
Comment (emilyjablonski) [inline src/helpers/MultiLineAddress.tsx:1]: We had a bug here with an extra space + some linting errors

File: src/navigation/ProgressNav.scss
Comment (emilyjablonski) [inline src/navigation/ProgressNav.scss:1]: Lots of small changes here to accommodate a core-styled version of the style updates in this Figma screenshot
Screenshot 2023-10-27 at 8 42 15 PM

File: src/page_components/ApplicationTimeline.scss
Comment (emilyjablonski) [inline src/page_components/ApplicationTimeline.scss:1]: This component overrides ProgressNav, so with some of my new changes I had to un-do them over here

**Review by ColinBuyck — APPROVED** [review]
LGTM 🏟️ appreciate the commenting

**Pull request merged**